app/webapp/src/shared/types/types.notifications.ts

```diff
@@ -0,0 +1,32 @@
+export enum ACTIVITY_EVENT_TYPE {
+  PostsParsed,
+  PostAutoposted,
+}
+
+export enum NOTIFICATION_FREQUENCY {
+  Instant,
+  Daily,
+  Weekly,
+  Monthly,
+  None,
+}
+
+export interface ActivityEvent {
+  id: string;
+  userId: string;
+  type: ACTIVITY_EVENT_TYPE;
+  timestamp: number;
+  involvedDocuments: Record<string, string[]>; // key is the collection name, value is the list of document ids
+  notified: boolean;
```

Comment:
I keep track of if an activity has been notified so I can query only for the unotified activities. An alternative approach would be to filter based on a time range (time since last notified I guess?)